### PR TITLE
MAM-3701-custom-emojis-not-appearing-in-activity-tab

### PR DESCRIPTION
**Pull request by rileyhCode**
- make postTextLabel a MetaText (MetaLabel would not allow limiting to 4 lines AFAICT)

### CODE REVIEW

File: Mammoth/Views/Cells/ActivityCardCell/ActivityCardCell.swift [hunk 1]
```diff
@@ -49,20 +50,44 @@ final class ActivityCardCell: UITableViewCell {
...
-        label.hashtagColor = .custom.highContrast
-        label.URLColor = .custom.highContrast
-        label.emailColor = .custom.highContrast
-        label.linkWeight = .semibold
-        label.isOpaque = true
-        label.urlMaximumLength = 30
-        label.numberOfLines = 4
-        label.translatesAutoresizingMaskIntoConstraints = false
-        return label
+    private var postTextLabel: MetaText = {
```
Comment (bnolens) [inline Mammoth/Views/Cells/ActivityCardCell/ActivityCardCell.swift:53]: I would use MetaLabel here instead, as we do in the PostCardCell. MetaLabel seems to be more performant in a TableViewCell. You can set the number of lines by setting `self.postTextView.textContainer.maximumNumberOfLines`

File: Mammoth/Views/Cells/ActivityCardCell/ActivityCardCell.swift [hunk 2]
```diff
@@ -229,17 +252,20 @@ private extension ActivityCardCell {
...
         
         contentStackView.addArrangedSubview(header)
-        contentStackView.addArrangedSubview(postTextLabel)
+        contentStackView.addArrangedSubview(postTextLabel.textView)
         contentStackView.addArrangedSubview(mediaContainer)
-                
-        let postTextTrailing = postTextLabel.trailingAnchor.constraint(equalTo: contentStackView.trailingAnchor)
-        postTextTrailing.priority = .defaultHigh
+        
+//        postTextLabel.linkDelegate = self
```
Comment (bnolens) [inline Mammoth/Views/Cells/ActivityCardCell/ActivityCardCell.swift:258]: Probably some wip work that can be removed?

File: Mammoth/Views/Cells/ActivityCardCell/ActivityCardCell.swift [hunk 3]
```diff
@@ -229,17 +252,20 @@ private extension ActivityCardCell {
...
+        postTextLabel.textView.linkDelegate = self
+
+        //        let postTextTrailing = postTextLabel.trailingAnchor.constraint(equalTo: contentStackView.trailingAnchor)
```
Comment (bnolens) [inline Mammoth/Views/Cells/ActivityCardCell/ActivityCardCell.swift:261]: Why did you have to remove this?

**Review by bnolens — CHANGES_REQUESTED** [review]
A few minor comments. Please have a look.

File: Mammoth/Views/Cells/ActivityCardCell/ActivityCardCell.swift (hunk 1 above)
Comment (rileyhCode) [reply Mammoth/Views/Cells/ActivityCardCell/ActivityCardCell.swift:53]: Weird; I missed this somehow. (I had switched to MetaLabel first, then went to MetaText…)

File: Mammoth/Views/Cells/ActivityCardCell/ActivityCardCell.swift (hunk 2 above)
Comment (rileyhCode) [reply Mammoth/Views/Cells/ActivityCardCell/ActivityCardCell.swift:258]: (left over from MetaLabel attempt)

File: Mammoth/Views/Cells/ActivityCardCell/ActivityCardCell.swift (hunk 3 above)
Comment (rileyhCode) [reply Mammoth/Views/Cells/ActivityCardCell/ActivityCardCell.swift:261]: Didn't need to.

**Review by bnolens — APPROVED** [review]
LGTM!

**Pull request merged**